### DISCUSSION TITLE
Release v0.8.0

### DISCUSSION THREAD

**Discussion by Z-Dante** [Z-Dante/Magpie#3] [Announcements]
### Version 0.8.0 is here, based on the v0.8.0 release of https://github.com/Blinue/Magpie/. This is a 1:1 release with no code changes except for the built-in scaling presets. The original changelog includes:
<details>
<summary>Show Original Changelog</summary>

v0.8 requires installation of the [.NET 6 Runtime](https://dotnet.microsoft.com/download)

**New Features**

1. Migration of user interface to .NET 6
2. Support for cropping the title bar of UWP windows. The title bar of some UWP windows is part of the user interface, so the option to disable this feature is provided
3. Support for multiple monitors
   * Support DPI scaling across multiple monitors
   * Support for full-screen windows spanning over multiple screens [Spanning over multiple screens #219](https://github.com/Blinue/Magpie/issues/219)
   * Cursor can't move freely between monitors when operating windows on other monitors [Cursor can't move freely between monitors #65](https://github.com/Blinue/Magpie/issues/65)
4. add new capture mode Desktop Duplication, requires OS win10 v2004 or newer 👉[Comparison of existing capture modes](https://github.com/Blinue/Magpie/wiki/%E6%8D%95%E8%8E%B7%E6%A8%A1%E5%BC%8F%E5%AF%B9%E6%AF%94)
5. Add the function of simulating exclusive fullscreen for a distraction-free effect [Simulate D3D exclusive fullscreen when zooming #245](https://github.com/Blinue/Magpie/issues/245)

**Effect change**

1. Renamed Mitchell to Bicubic, parameters are no longer compatible
2. Added NIS and NVSharpen, see [v0.8 plan #202 (comment)](https://github.com/Blinue/Magpie/issues/202#issuecomment-981352269) for examples of their use
3. Add some CRT shaders, including CRT-Easymode, CRT-Geom, CRT-Hyllian, CRT-Lottes and GTU [CRT emulation shader #160](https://github.com/Blinue/Magpie/issues/160)
4. add Anime4K_Thin_HQ and Anime4K_Upscale_GAN_x2_S
5. Add LumaSharpen and FineSharp [Request to add several sharpening filters #242](https://github.com/Blinue/Magpie/issues/242)
6. Add FSR_EASU_DX10, which is FSR_EASU with DirectX feature level 10 support, slightly slower than the original version

Thanks to @hooke007 for maintaining a sample configuration of all the effects of Magpie [gist.github.com/hooke007/818ecc88f18e229bca743b7ae48947ad](https://gist.github.com/hooke007/818ecc88f18e229bca743b7ae48947ad)

**Other changes**

1. Optimized the compatibility of Graphics Capture mode
2. Optimized frame rate visibility on white backgrounds [(Request) Shadow for fps indicator #111](https://github.com/Blinue/Magpie/issues/111)
3. It is now safe to launch Magpie in other programs/scripts without setting a working path [Unable to scale games when launching Magpie with 3rd party front-end launcher #236](https://github.com/Blinue/Magpie/issues/236)

**System requirements**

1. Windows 10/11
2. DirectX feature level 10
3. [.NET 6 Runtime](https://dotnet.microsoft.com/download)
4. [MSVC 2015-2022 Runtime](https://docs.microsoft.com/cpp/windows/latest-supported-vc-redist)

**Caution** 

1. v0.8 is the last version to support DirectX feature level 10
2. Magpie's implementation of NIS is much slower than FSR, but NVSharpen is quite fast
3. RTSS Fails to hook this version of Magpie

_Translated with www.DeepL.com/Translator (free version)_

</details>

## _**Scaling Preset changes:**_ 

Lots  of changes to the scaling presets since the last release. Changes include:

1. **Inclusion of CRT presets** for the retro enthusiasts: CRT Cathode, CRT Easymode, CRT Geom and CRT Hylian. 
	_Some Comparison of CRT shaders (Scaled to 1080p):_     
	- _Kichikuou Rance : https://imgsli.com/ODY5MjE/0/1_      
	- _Ikusa Megami Zero: https://imgsli.com/ODY5MjU/0/1_      
	- _CPS1 Cadillacs and Dinosaurs: https://imgsli.com/ODcwMzk/0/1_     
2. **Experimental Anime4kGAN shader preset**. Much better in keeping details compared to ACNet and regular Anime4k. Produces crisp and sharp image with minimal detail loss.
	- _White Album 2 Anime4kGAN 720p -> 1080p: https://imgsli.com/ODczODg/1/3_
3. **Anime4k Thinlines preset**. Thins out the lines of anime artworks after upscaling. Might look better in some cases and worse in others. 
	- _Duel Savior 480p->1080p: https://imgsli.com/ODc2MTE/1/2_
4. **Anime4kGAN Restore**. Filters out the noises created by the GAN shader while keeping all the nice details. 
	- _Kunado Kokuki 720p -> 1080p: https://imgsli.com/ODc1NTQ/1/5_
	- _Duel Savior 480p->1080p: https://imgsli.com/ODc2MTE/1/5_
5. **Anime4k 3x / 3x Light**. Experimental stack of Anime4kGan + Restore + Upscale Light shaders to upscale images for high resolution displays. Should only be used for 1440p/4k displays.

For editing and configuring your own scaling presets please check out https://github.com/Z-Dante/Magpie/blob/master/docs/CustomScaleConfiguration.md

### Full list of scaling presets in this release of Magpie:
<details>
<summary>Show Preset details</summary>

The following table contains the list of all the scaling presets presents in this release of Magpie. It also shows their GPU usage, Power usage and Temperature taken during a 720p -> 1080p upscaling test using the GDI API.  Benchmark done one a PC running Windows 11 and using a RTX 2060 GPU. Results will differ for different configurations. 

<table>
<tr>
	<td>Preset Name
	<td>GPU Usage
	<td>Power Usage
	<td>Temperature
<tr>
	<td>NONE
	<td>25%
	<td>18W
	<td> 34°C
<tr>
	<td>ACNet
	<td>38%
	<td>44W
	<td> 37°C
<tr>
	<td>ACNet (RCAS)
	<td>38%
	<td>44W
	<td> 37°C
<tr>
	<td>Anime4k
	<td>39%
	<td>43W
	<td> 38°C
<tr>
	<td>Anime4k Thinlines
	<td>51%
	<td>50W
	<td> 38°C
<tr>
	<td>Anime4kGAN (S)
	<td>37%
	<td>50W
	<td> 39°C
<tr>
	<td>Anime4kGAN (S) Restore Soft
	<td>53%
	<td>60W
	<td> 40°C
<tr>
	<td>ANime4kGAN (S) Restore VL !
	<td>73%
	<td>150W
	<td> 68°C
<tr>
	<td>FSRCNNXLineArt
	<td>36%
	<td>44W
	<td> 40°C
<tr>
	<td>FSR
	<td>24%
	<td>24W
	<td> 39°C
<tr>
	<td>Lanczos
	<td>35%
	<td>27W
	<td> 38°C
<tr>
	<td>Anime4K (Light)
	<td>36%
	<td>26W
	<td> 37°C
<tr>
	<td>Anime4K (Very Light)
	<td>37%
	<td>25W
	<td> 37°C
<tr>
	<td>ANime4k Denoise + FSR
	<td>38%
	<td>41W
	<td> 39°C
<tr>
	<td>CRT Hylian
	<td>20%
	<td>22W
	<td> 36°C
<tr>
	<td>CRT Easymode
	<td>38%
	<td>22W
	<td> 36°C
<tr>
	<td>CRT Geom
	<td>38%
	<td>23W
	<td> 36°C
<tr>
	<td>CRT Cathode
	<td>29%
	<td>26W
	<td> 36°C
<tr>
	<td>Anime4k 3x !
	<td>69%
	<td>143W
	<td> 68°C
<tr>
	<td>Anime4k 3x Light
	<td>58%
	<td>64W
	<td> 47°C
</table>

\* Anime4k Native and Anime4k Native light were tested at 4k. Everything else tested at 1080p; The results should be similar for 1440p too since most of these scalers scales to 1440p before downscaling to 1080p. 
\*\* The presets marked with ! are extremely heavy and not recommended for low end GPUs
</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/Z-Dante/Magpie/releases/tag/v0.8.0'>Release v0.8.0</a>.</em>

**Comment by it0na72** [comment]
Thanks for this amazing update! 
Even though multiple screens now work, when I try to take a screenshot with ShareX (ctrl+4 for example), the game goes back to window mode. Is this normal?

**Reply by Z-Dante** [reply]
Which capture mode are you using? Screenshots do not work in "Desktop Duplication" mode. However ShareX worked just fine with all the other 3 capture methods. I tried with both full-screen capture and region capture options from shareX. I do not have multiple screens either so can't say anything about multi-screen specific problems. 

I'm not sure what ctrl+4 does. But if it's the hotkey for Region Capture, then it switches to window mode for me too. I suggest you to enable the "Restore Fullscreen When Source Window Regain Focus" from options as an workaround for now. 

If you believe this is an actual issue, then please make an issue / comment about this problem over at https://github.com/Blinue/Magpie/

**Reply by it0na72** [reply]
That works, thanks for the detailed response. 
Thank you so much!

**Comment by Rroork** [comment]
+1 thank you for your hard work!  For clarification, Anime4kGAN Restore is better than Anime4kGAN?

**Reply by Z-Dante** [reply]
Depends entirely on the source. Restore applies a fair bit of denoising and sharpening effects so it might look better in some cases and worse in others. Just try both and stick to whichever one you like. I'd recommend Anime4kGAN over restore for most games.

**Comment by davidmwest** [comment]
Anime4k 3x is beautiful and by far the best on my 1440p display, any particular reason it was removed from v0.9.0?

**Reply by Z-Dante** [reply]
Ah. The preset menu was getting too cluttered so I removed some of the presets to make things less confusing 😅 
The preset is still there in the ScaleModels.json file, just commented out.  
Extract and copy the ScaleModels.json files to magpie folder and it should be back

[ScaleModels.zip](https://github.com/Z-Dante/Magpie/files/8968488/ScaleModels.zip)
